src/docs/Architecture.md

```diff
@@ -0,0 +1,32 @@
+## General Moments
+
+### Everything is a Knowledge Item. 
+#### KnowledgeItem 
+KnowledgeItem - abstract class, that is parental for all other items
+**Unique fields: _title, creation_time,  id: **
+
+
+### Information Classes 
+These classes, such as TextNote, Comment, etc. are inherited from KnowledgeItem. Their main function – to store a specific type of information. 
```

Comment:
In future, please avoid renaming files/directories and changing content of a file in a single commit. Such commits are harder to review, and git can't understand that the file was renamed. Rename without any changes to content should be a separate commit, changes to content as one or more commits.

Reply:
Understood, I thought git can track file moving and renaming

Reply:
git can track file moving/renaming, but it uses some heuristics to distinguish between rename and remove/add - so this works only if the changes to the file are minimal.  You can compare how different commits are displayed: f48822b12038de46f39b1782ec36e1646e140519  shows rename, while 0b684d443daf55ac7e6276a7702ce91b8d265838 shows one file deleted and another unrelated file created

So it is best practice to rename/move file as one commit (without any changes to content), and then change content in a separate commit